File: update_json.js
const fs = require('fs');
const path = require('path');

try {
    const filePath = path.join(__dirname, 'pages/json/chordProgression.json');
    const jsonStr = fs.readFileSync(filePath, 'utf8');
    const data = JSON.parse(jsonStr);

    // Add youtubeId and clipStart to all music entries
    function processData(obj) {
        if (Array.isArray(obj)) {
            obj.forEach(item => processData(item));
        } else if (obj && typeof obj === 'object') {
            // Check if this is a music entry
            if (obj.genre && obj.artist && obj.title) {
                if (!obj.youtubeId) obj.youtubeId = '';
                if (!obj.clipStart) obj.clipStart = 0;
            }
            // Recursively process nested objects
            for (const key in obj) {
                if (obj.hasOwnProperty(key)) {
                    processData(obj[key]);
                }
            }
        }
    }

    processData(data);
    
    fs.writeFileSync(filePath, JSON.stringify(data, null, 2) + '\n', 'utf8');
    console.log('Successfully added youtubeId and clipStart fields to all music entries');
} catch (err) {
    console.error('Error:', err.message);
    process.exit(1);
}

File: update_json.py
#!/usr/bin/env python3
import json

with open('pages/json/chordProgression.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Process all progressions and their music entries
def add_youtube_fields(obj):
    if isinstance(obj, dict):
        # If this is a music entry with genre field, add youtubeId and clipStart
        if 'genre' in obj and 'artist' in obj and 'title' in obj:
            if 'youtubeId' not in obj:
                obj['youtubeId'] = ''
            if 'clipStart' not in obj:
                obj['clipStart'] = 0
        # Recursively process all values
        for value in obj.values():
            add_youtube_fields(value)
    elif isinstance(obj, list):
        for item in obj:
            add_youtube_fields(item)

add_youtube_fields(data)

with open('pages/json/chordProgression.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)
    f.write('\n')

print('Successfully updated JSON with youtubeId and clipStart fields')
